packages/redux/src/reducers/content.ts

```diff
@@ -36,11 +50,175 @@ export function itemsReducer(state = {
   }
 }
 
-export function navigationReducer(state = {
-  loading: {},
+var currentDescriptorUrl;
+export function descriptorsReducer(state = {
+  loading: {}, // { all: boolean, [id: string]: boolean }
   entries: {}
-}, action: AnyAction) {
+}, action: AnyAction): StateContainer<Item> {
   switch (action.type) {
+    case GET_DESCRIPTOR: {
+      currentDescriptorUrl = action.payload;
+
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [currentDescriptorUrl]: true
+        }
+      }
+    }
+    case GET_DESCRIPTOR_COMPLETE: {
+      const descriptor = action.payload;    //TODO: Properly set Descriptor model
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [currentDescriptorUrl]: false
+        },
+        entries: {
+          ...state.entries,
+          [currentDescriptorUrl]: descriptor
+        }
+      }
+    }
+    default:
+      return state
+  }
+}
+
+var currentChildrenUrl;
+export function childrenReducer(state = {
+  loading: {}, // { all: boolean, [id: string]: boolean }
+  entries: {}
+}, action: AnyAction): StateContainer<Item> {
+  switch (action.type) {
+    case GET_CHILDREN: {
+      currentChildrenUrl = action.payload;
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [currentChildrenUrl]: true
+        }
+      }
+    }
+    case GET_CHILDREN_COMPLETE: {
+      const item: Item = action.payload;
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [currentChildrenUrl]: false
+        },
+        entries: {
+          ...state.entries,
+          [currentChildrenUrl]: item
+        }
+      }
+    }
+    default:
+      return state
+  }
+}
+
+export function treeReducer(state = {
+  loading: {}, // { all: boolean, [id: string]: boolean }
+  entries: {},
+  childIds: {}
+}, action: AnyAction): StateContainer<any> {
+  switch (action.type) {
+    case GET_TREE: {
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [action.payload.url]: true
+        }
+      }
+    }
+    case GET_TREE_COMPLETE: {
+      const flatEntries = flattenEntries(action.payload, 'children');
```

Comment:
'children' is already the default property name on the method. no need to supply it unless different.